src/platform.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018 Joe Wilm, The Alacritty Project Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//! Provide operating system and platform-level integration.
+#[cfg(windows)]
+extern crate winapi;
+#[cfg(windows)]
+use std::panic;
+#[cfg(windows)]
+use std::{ffi, ptr};
+#[cfg(windows)]
+use winapi::um::winuser;
+
+#[cfg(windows)]
+fn on_windows_error(pi: &panic::PanicInfo) {
+    let msg = format!("{}\nPress Ctrl-C to Copy", pi);
+    unsafe {
+        winuser::MessageBoxA(
+            ptr::null_mut(),
+            ffi::CString::new(msg).unwrap().as_ptr(),
+            ffi::CString::new("Alacritty: Runtime Error")
+                .unwrap()
+                .as_ptr(),
+            winuser::MB_OK | winuser::MB_ICONERROR,
+        );
+    }
+}
+
+// Install a panic handler that renders the panic in a classical Windows error
+// dialog box.
+#[cfg(windows)]

```

Comment:
Loads of duplicate `#[cfg(windows)]` annotations everywhere. I'd recommend just using this single annotation and renaming the function to `set_panic_hook` (or similar), then you can simply provide an empty function for everything non-windows and you'll only need the annotations once right here.

Reply:
Pardon me, since I am very new to Rust.  Is it possible to group annotations at all?  Also, would you just suggest inlining the use statements into this function as well?

Reply:
OK.  I think the current revision captures your feedback in entirety.  Please let me know if it doesn't.